test/test_sparse.py

```diff
@@ -985,7 +985,10 @@ def test_factory_size_check(self):
     def test_factory_empty_indices(self):
         device = 'cuda' if self.is_cuda else 'cpu'
         tensor = torch.sparse_coo_tensor([], [], torch.Size([]), device=device)
-        expected_indices = torch.tensor([], dtype=torch.long, device=device)
+        if torch._C._use_zero_size_dim():
+            expected_indices = torch.empty((1, 0), dtype=torch.long, device=device)
```

Comment:
what about functions non-constructors that can return empty sparse tensors?  Do those do the right things?  What about if it isn't the simple (1, 0) case -- but something more complicated (more dimensions, more 0s, etc?).

Reply:
I added more test cases for more dimensions / more 0s. For functions non-constructors that can return empty sparse tensors, I think they will always use the constructor to create that empty sparse tensor, which will then do the right thing.